src/user/model.rs

```diff
@@ -1,30 +1,33 @@
 use crate::schema::*;
+use crate::user::util::{make_hash, make_salt};
 use chrono::*;
+use shrinkwraprs::Shrinkwrap;
 use uuid::Uuid;
-extern crate rand;
-use crate::user::util::{make_hash, make_salt};
 
 #[derive(Debug, Serialize, Deserialize, Queryable, juniper::GraphQLObject)]
 pub struct User {
     pub user_id: i32,
     pub user_uuid: Uuid,
     #[graphql(skip)]
     pub hash: Vec<u8>,
+    #[graphql(skip)]
     pub salt: String,
     pub email: String,
-    pub created_at: NaiveDateTime,
+    pub role: String,
     pub name: String,
+    pub created_at: NaiveDateTime,
 }
 
 #[derive(Debug, Insertable)]
 #[table_name = "users"]
-pub struct UserInsert {
```

Comment:
`UserInsert` sounds like an action to me, so I preferred `InsertableUser` which is an "object" to me.